uftasks/uftasks-showcase/uftasks-webapp/src/main/java/org/uberfire/client/screens/popup/NewFolderView.java

```diff
@@ -68,16 +79,22 @@ public void show() {
     @Override
     public void hide() {
         modal.hide();
-        folderNameTextBox.setText( "" );
+        folderNameTextBox.setValue( "" );
     }
 
-    @EventHandler("ok-button")
-    public void addFolder( ClickEvent event ) {
-        presenter.newFolder( folderNameTextBox.getText() );
+    @SinkNative( Event.ONCLICK )
+    @EventHandler( "ok-button" )
+    public void addFolder( Event event ) {
+        presenter.newFolder( folderNameTextBox.getValue() );
     }
 
-    @EventHandler("cancel-button")
-    public void cancel( ClickEvent event ) {
+    @EventHandler( "cancel-button" )
+    public void cancel( Event event ) {
         presenter.close();
     }
+
+    @Override
+    public HTMLElement getElement() {
+        return view;
```

Comment:
Also (and I can see that it was already like this, but maybe it would be good to fix it now) I don't think it's good for a `@Templated` bean to add it's root element as a child of something internally. It probably isn't a problem now, but if anyone ever tried injecting this view as a `@DataField` weird things would happen. It would be cleaner to separate this into to parts: the templated part and the modal part. The modal view would not be `@Templated` and would be a view that injects the templated view and adds it to the modal.